src/macaron/slsa_analyzer/package_registry/package_registry.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023 - 2023, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module defines package registries."""
+
+import logging
+from abc import ABC, abstractmethod
+
+from macaron.slsa_analyzer.build_tool.base_build_tool import BaseBuildTool
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class PackageRegistry(ABC):
+    """Base package registry class."""
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    @abstractmethod
+    def load_defaults(self) -> None:
+        """Load the .ini configuration for the current package registry."""
+
+    @abstractmethod
+    def is_detected(self, build_tool: BaseBuildTool) -> bool:
+        """Detect if the package registry is compatible with the given build tool.
```

Comment:
The doc string is a bit ambiguous for me (it says `compatible` but the method name is `is_detected`).

Reply:
The naming of the `is_detected` method here is to be consistent with the methods `is_detected` in other classes (e.g. git services, build tools). I should have written the docstring more carefully though. The docstring is fixed in d0a8ecaf372998213fa8fd69bcb80b7c5ab4b79a.